tests/tests.py

```diff
@@ -215,19 +215,42 @@ def __init__(self):
                 # [7-8] python version A=0, B=1,..
                 # [9] A,B -- we may need more than one project
 
-                prefix = 'Z' + (re.sub("[^A-Z]", "",
-                                       getpass.getuser().upper()))[0:6] + \
-                         str(sys.version_info[0]) + \
-                         str(sys.version_info[1])
+                """ `jid` is important for avoiding concurency problems when
+                executing tests in parallel as we have only one test instance.
+
+                jid lenght must be less than 9 characters because we may append
+                another one and the JIRA Project key length limit is 10.
+
+                Tests run in parallle:
+                * git branches master or developer, git pr or developers running
+                  tests outside Travis
+                * Travis is using "Travis" username
+
+                https://docs.travis-ci.com/user/environment-variables/
+                """
+                user = re.sub("[^A-Z_]", "", getpass.getuser().upper())
 
-                self.project_a = prefix + 'A'  # old XSS
+                def hashify(some_string, max_len=8):
+                    return hashlib.md5(some_string.encode('utf-8')).hexdigest()[:8].upper()
+
+                if user == 'TRAVIS' and 'TRAVIS_JOB_NUMBER' in os.environ:
+                    # please note that user underline (_) is not suppored by
+                    # jira even if is documented as supported.
+                    self.jid = 'T' + hashify(user + os.environ['TRAVIS_JOB_NUMBER'])
```

Comment:
I guess we don't need to bring travis user into it if we're namespacing the jid with T for travis and Z for others?

Also, TRAVIS_JOB_NUMBER is going to be something like 531.3 - why do we even need to hash this to 8 digits? Can't we just replace the '.' and have a much more human-readable project key?